glusterd2/config.go

```diff
@@ -126,20 +145,28 @@ func initConfig(confFile string) error {
 	// Limit config to toml only to avoid confusion with multiple config types
 	config.SetConfigType("toml")
 
-	// If custom configuration is passed
+	confFile := config.GetString("config")
+	// If custom configuration is passed use it, if not try to use defaults
 	if confFile != "" {
 		config.SetConfigFile(confFile)
-		if err := config.MergeInConfig(); err != nil {
+	} else {
+		config.AddConfigPath(path.Join(defaultPathPrefix, "/etc/glusterd2"))
+		config.SetConfigName("glusterd2")
+	}
+	if err := config.ReadInConfig(); err != nil {
+		// Ignore error if config file is not found, error out otherwise
+		if _, ok := err.(config.ConfigFileNotFoundError); ok {
 			log.WithError(err).
-				WithField("file", confFile).
-				Error("failed to read config file")
+				WithField("file", config.ConfigFileUsed()).
+				Warn("failed to load config from file")
```

Comment:
both error message is same.

Reply:
One is a warning the other is an error. If a config file is found and we cannot parse it, it is an error. If we don't find a config file at all, we don't care.